plugins/qtAccessionPlugin/lib/model/QubitAccession.php

```diff
@@ -26,10 +26,10 @@ public function __toString()
 
   protected function insert($connection = null)
   {
-    // If no identifier has been specified, use next available one
-    if (!$this->identifier)
+    // If identifier has been specified and the mask is enabled, increment the counter
+    if (!empty($this->identifier) && QubitAccession::maskEnabled())
     {
-      $this->identifier = self::nextAvailableIdentifier();
+      self::incrementAccessionCounter();
```

Comment:
We may need to wrap the `incrementAccessionCounter` function content in a transaction too as it may be called from CLI in here after this changes, and the transaction in `nextAvailableIdentifier` won't be working in that case.